json-logic/schema.ts

```diff
@@ -0,0 +1,146 @@
+// LogicExpression refers to the JSONLogic standard.
+import { LogicExpression } from "./logic";
+
+/*
+* This module defines a schema for questionaire and result calulation logic.
+*/
+
+/**
+ * Type of a question, essentially defines the result type.
+ */
+export enum QuestionType {
+	Select = "Select",
+	Multiselect = "Multiselect",
+	Number = "Number",
+	Boolean = "Boolean",
+	Date = "Date",
+	Text = "Text"
+}
+
+/**
+ * Option for multi-select questions.
+ */
+export interface IOption {
+	/**
+	 * Human-Readable answer, can be localized.
+	 */
+	text: string,
+	/**
+	 * Value used for evaluating logic expressions.
+	 */
+	value: string;
+}
+
+/**
+ * Represents a question.
+ */
+export interface IQuestion {
```

Comment:
I assume that we need some categories and some scoring to make the logic easier. How about this:

1. questions get one or more categories
2. each answer option gets a score (at least single- and multi-selects), which is added to all categories of the question

The implementation has to keep track of this score, which can be used in logic expressions.

Alternatively, we could also attach the categories to the answer options, but I think it is cleaner if they belong to the question. One implication: a multi-select of respiratory and non-respiratory symptoms would need to be done in two separate questions. But I think that that is okay.

Reply:
Or we don't do multiselects :)

Reply:
I would agree with Daniel.  If it's not absolutely necessary, I'd also try to avoid muliselects to simplify the logic.